components/History/History.tsx

```diff
@@ -45,42 +57,104 @@ const History: React.FunctionComponent<HistoryProps> = ({
-  const { translate, transactions, language, setBackgroundError, addLastSnackbar } = context;
+  const { translate, valueTransfers, language, setBackgroundError, addLastSnackbar } = context;
   const { colors } = useTheme() as unknown as ThemeType;
   moment.locale(language);
 
-  const [isTxDetailModalShowing, setTxDetailModalShowing] = useState<boolean>(false);
-  const [txDetail, setTxDetail] = useState<TransactionType>({} as TransactionType);
-  const [numTx, setNumTx] = useState<number>(50);
-  const [loadMoreButton, setLoadMoreButton] = useState<boolean>(numTx < (transactions.length || 0));
-  const [transactionsSorted, setTransactionsSorted] = useState<TransactionType[]>([]);
+  const [isValueTransferDetailModalShowing, setValueTransferDetailModalShowing] = useState<boolean>(false);
+  const [valueTransferDetail, setValueTransferDetail] = useState<ValueTransferType>({} as ValueTransferType);
+  const [valueTransferDetailIndex, setValueTransferDetailIndex] = useState<number>(-1);
+  const [numVt, setNumVt] = useState<number>(50);
+  const [loadMoreButton, setLoadMoreButton] = useState<boolean>(numVt < (valueTransfers.length || 0));
+  const [valueTransfersSorted, setValueTransfersSorted] = useState<ValueTransferType[]>([]);
+  const [isAtTop, setIsAtTop] = useState<boolean>(true);
+  const scrollViewRef = useRef<ScrollView>(null);
+
+  useScrollToTop(scrollViewRef);
 
   var lastMonth = '';
 
-  const fetchTransactionsSorted = useMemo(() => {
-    return transactions.sort((a, b) => b.time - a.time).slice(0, numTx);
-  }, [transactions, numTx]);
+  const fetchValueTransfersSorted = useMemo(() => {
+    // we need to sort the array properly.
+    // by:
+    // - time
+    // - txid
+    // - address
+    // - pool
+    return valueTransfers
+      .sort((a: ValueTransferType, b: ValueTransferType) => {
+        const timeComparison = b.time - a.time;
+        if (timeComparison === 0) {
+          // same time
+          const txidComparison = a.txid.localeCompare(b.txid);
+          if (txidComparison === 0) {
+            // same txid
+            const aAddress = a.address?.toString() || '';
+            const bAddress = b.address?.toString() || '';
+            const addressComparison = aAddress.localeCompare(bAddress);
+            if (addressComparison === 0) {
+              // same address
+              const aPoolType = a.poolType?.toString() || '';
+              const bPoolType = b.poolType?.toString() || '';
+              // last one sort criteria - poolType.
+              return aPoolType.localeCompare(bPoolType);
+            } else {
+              // different address
+              return addressComparison;
+            }
+          } else {
+            // different txid
+            return txidComparison;
+          }
+        } else {
+          // different time
+          return timeComparison;
+        }
+      })
+      .slice(0, numVt);
+  }, [valueTransfers, numVt]);
```

Comment:
i am not familiar with this code but does this only sort the value transfer or also combine them in some cases?

Reply:
the `slice` cut the first `numVt` lines... the list in the beginning have the first 50 lines... and you can tap on `load more` until 100, and so on...